packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -373,17 +373,17 @@ class DevFS {
 
     // Handle deletions.
     printTrace('Scanning for deleted files');
-    String assetBuildDirPrefix = getAssetBuildDirectory() + fs.path.separator;
-    final List<String> toRemove = new List<String>();
-    _entries.forEach((String devicePath, DevFSContent content) {
+    String assetBuildDirPrefix = fs.path.toUri(getAssetBuildDirectory()).path + '/';
```

Comment:
We have to be careful: This is not really a round trip. We are converting a platform-**specific** file path (`getAssetBuildDirectory()`) to a platform-**independent** Uri file path. While these look similar on some platforms (POSIX) those two can be different on other platforms (e.g. Windows uses "\\" as platform-specific path separator, whereas platform-independent Uris use "/").